src/inject.ts

```diff
@@ -1,36 +1,50 @@
-import React, { createElement } from "react"
+import * as React from "react"
 import { observer } from "./observer"
 import { copyStaticProperties } from "./utils/utils"
-import { MobXProviderContext } from "./Provider"
+import { MobXProviderContext, TProviderShape } from "./Provider"
+import { IReactComponent } from "./index"
+
+export type IValueMap = Record<string, any>
+export type IStoresToProps<
+    S extends IValueMap = {},
+    P extends IValueMap = {},
+    I extends IValueMap = {},
+    C extends IValueMap = {}
+> = (stores: S, nextProps: P, context?: C) => I
 
 /**
  * Store Injection
  */
-function createStoreInjector(grabStoresFn, component, injectNames, makeReactive) {
+function createStoreInjector(
+    grabStoresFn: IStoresToProps,
+    component: IReactComponent<any>,
+    injectNames: string,
+    makeReactive: boolean
+): IReactComponent<any> {
     // Support forward refs
-    let Injector = React.forwardRef((props, ref) => {
-        const newProps = { ...props }
-        const context = React.useContext(MobXProviderContext)
+    let Injector: IReactComponent<any> = React.forwardRef((props, ref) => {
```

Comment:
```suggestion
    let Injector = React.forwardRef((props, ref) => {
```

Reply:
Removing this will cause the compiler to complain on line 37 `if (makeReactive) Injector = observer(Injector)`, because forwardRef returns a ForwardRefExoticComponent.  I can either leave it as it is, or cast `observer(Injector)` as any.  Up to you.

Reply:
I see, ok let's leave it like this.